### PR TITLE
Add support for spot instances

### PR DESCRIPTION
**Pull request by MarcelMue**
Please make sure in case you are changing the spec all our customers are informed beforehand.

You can throw a message in #sig-customer with the changes and the Solution Engineers will take care of
sharing the changes with the customers.

### CODE REVIEW

**Comment by marians** [comment]
I would prefer plural `spot_instances_enabled` over singular `spot_instance_enabled`, since a node pool is normally about a group of instances.

File: spec/definitions.yaml [hunk 1]
```diff
@@ -928,14 +928,27 @@ definitions:
...
           Attributes specific to the AWS provider
         properties:
+          instance_distribution:
+            description: |
+              Attributes defining the instance distribution in a node pool.
+            type: object
+            properties:
+              on_demand_base_capacity:
+                description: |
+                  Base capacity of on demand machines.
```
Comment (marians) [inline spec/definitions.yaml]: The description should explain what exactly the number would mean. Is it a percentage or a number of machines?

File: spec/definitions.yaml [hunk 2]
```diff
@@ -928,14 +928,27 @@ definitions:
...
+              on_demand_percentage_above_base_capacity:
+                description: |
+                  Percentage of on demand instances above the base capacity.
+                type: integer
           instance_type:
             type: string
             description: |
               EC2 instance type to use for all nodes in the node pool. _(Validated against available instance types.)_
-          spot_instance_enabled:
+          use_alike:
```
Comment (marians) [inline spec/definitions.yaml]: I would like this to be more specific, e. g. `use_alike_instance_types`.

**Comment by marians** [comment]
Actually, I don't feel as strongly after some more thinking. However, the comment

```
# Specification of a node in a node pool to be created
V5AddNodePoolRequestNodeSpec:
```

should be adapted to reflect the actual purpose better.

**Comment by marians** [comment]
Related to https://github.com/giantswarm/giantswarm/issues/8347

File: spec/definitions.yaml [hunk 3]
```diff
@@ -1000,12 +1037,21 @@ definitions:
         description: Information on the current size and status of the node pool
         type: object
         properties:
+          instance_types:
```
Comment (MarcelMue) [inline spec/definitions.yaml]: @marians I couldn't think of a good structure here which doesn't totally change this stuff. Do you have an idea?

**Comment by marians** [comment]
Consistency thing: property names should use camel case. E. g. `instance_distribution` should be `instanceDistribution`.

File: spec/definitions.yaml (hunk 3 above)
Comment (marians) [reply spec/definitions.yaml:1040]: This mention went past me, sorry. Don't see the problem here actually.

**Comment by marians** [comment]
FYI: With the latest commits I also added `default` attributes to inform users about what we set as a default value if the attribute is not specified when creating a node pool.

**Pull request merged**